### DISCUSSION TITLE
osu!gauntlet

### DISCUSSION THREAD

**Discussion by iSimpFor** [ppy/osu#16664] [Ideas] [answered]
the idea is to challenge the player to surviving a bunch of beatmaps while on limited life

unlike a dan or a pack, where the maps are the same in each, a gauntlet challenges you with a random set of ranked maps

you start with 100 hp(max) and lose 5 from missing and 3 from getting a meh/50, you do not recover it between beatmaps nor can you recover it during play(hp drain is disabled)
when you clear a map in the gauntlet, you play the next map as soon as you load the map or after 3 seconds, whichever takes longer

the game downloads and holds maps temporarily if you don't have them when you start to play the gauntlet

when you pass or fail the gauntlet, the results screen shows the total judgements and what beatmaps you played(and would've played if you didn't fail), you get the option to keep the maps you haven't downloaded, anything you didn't choose to keep will be discarded after you exit the game

there are two types of gauntlets: local and ranked

ranked gauntlets randomly pick a certain amount of ranked maps with certain requirements to create a well rounded challenge

there will be a gauntlet for certain star ratings

here is a gauntlet table(osu!standard):
(all maps picked will be random and ranked)
- 1* gauntlet: 4 maps between 1* and 2*
- 2* gauntlet: 4 maps between 2* and 3*
- 3* gauntlet: 5 maps between 3* and 4*, one should be skewed towards aim and one should be skewed toward speed
- 4* gauntlet: 5 maps between 4* and 4.5*, one should be skewed towards aim and one toward speed
- 4.5* gauntlet: 6 maps between 4.5* and 5*, one should be skewed towards aim, one toward speed, and one should be either aim; balanced; or speed
- 5* gauntlet: 6 maps between 5* and 5.5*, one should be skewed towards aim, one toward speed, and one should be either aim; balanced; or speed(at random)
- 5.5* gauntlet: 7 maps between 5.5* and 6*, one should be skewed towards aim, one toward speed, and two should be either aim; balanced; or speed(at random)
- 6* gauntlet: 7 maps between 6* and 6.5*, two should be skewed towards aim, two toward speed, and two should be either aim; balanced; or speed(at random)
- 6.5* gauntlet: 8 maps between 6.5* and 7*, two should be skewed towards aim, two toward speed, and two should be either aim; balanced; or speed(at random)
- 7* gauntlet: 8 maps between 7* and 7.5*, two should be skewed towards aim, two toward speed, and two should be either aim; balanced; or speed(at random)
- any gauntlet above the 7* gauntlet will give 9 maps between the mentioned star rating and 0.5* above that, two-three should be skewed towards aim, two-three toward speed, and two should be either aim; balanced; or speed(at random)

a gauntlet at a certain level should only appear once there are 100 or more maps(suggested value) around said star rating
you should also be able to have to option to only pick featured artist maps(the above requirement for a gauntlet still applies but with featured artist maps only)

the total maps length should equal or be above 1:45*the amount of maps picked originally, if it's under said value, the game will randomly pick maps at said star rating until the length requirement is satisfied 

this gauntlet type could give achievements for passing a gauntlet

for local gauntlets:
it's the same thing as a ranked gauntlet except any maps in the local copy of osu can be picked instead(ignoring maps that are hidden)

this gauntlet type doesn't give achievements

**Comment by hlysine** [comment]
The changes to HP could be made into a mod. I imagine it would be fun for long maps.

As for the rest of the gauntlet (map selection, scoring, etc), it's so different from the normal osu I feel like it should exist as a custom ruleset.

**Comment by peppy** [comment, marked as answer]
A similar mode is already planned for playlist mode, where you can set the play style to "marathon" and the user will need to play through all beatmaps in the order specified without break.

**Reply by iSimpFor** [reply]
i almost forgot to ask but will there be offline implementation?
